marklogic-data-hub/src/main/java/com/marklogic/hub/job/JobDataManager.java

```diff
@@ -0,0 +1,66 @@
+package com.marklogic.hub.job;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.dataservices.ExecCaller;
+import com.marklogic.client.dataservices.IOEndpoint;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.hub.HubConfig;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class JobDataManager {
+    ObjectNode apiObj;
+    String apiPath = "/ml-modules/root/data-hub/5/data-services/bulk/deleteJobs.api";
+    ObjectMapper mapper = new ObjectMapper();
+    HubConfig hubConfig;
+
+    JobDataManager(HubConfig hubConfig) {
+        this.hubConfig = hubConfig;
+        try (InputStreamReader apiReader = new InputStreamReader(this.getClass().getResourceAsStream(apiPath))) {
+            apiObj = mapper.readValue(apiReader, ObjectNode.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void cleanJobsData(String retainDuration) {
+        // See https://stackoverflow.com/a/52645128/535924 for more information about the duration regex
+        if (retainDuration == null || !retainDuration.matches("^(-?)P(?=.)((\\d*)Y)?((\\d+)M)?((\\d*)D)?(T(?=\\d)(\\d+H)?(([\\d]+)M)?(([\\d]+(?:\\.\\d+)?)S)?)?$")) {
+            throw new IllegalArgumentException("retainDuration must be a duration in the format of PnYnM or PnDTnHnMnS.");
+        }
+        DatabaseClient jobsClient = hubConfig.newHubClient().getJobsClient();
+        ObjectNode endpointConstants = mapper.createObjectNode().put("batchSize", 250).put("retainDuration", retainDuration);
+        ExecCaller endpoint = ExecCaller.on(jobsClient, new JacksonHandle(apiObj));
+        CapturingErrorListener errorListener = new CapturingErrorListener();
+
+        ExecCaller.BulkExecCaller bulkCaller = endpoint.bulkCaller(endpoint.newCallContext()
+                .withEndpointConstantsAs(endpointConstants));
+        bulkCaller.setErrorListener(errorListener);
+        bulkCaller.awaitCompletion();
+
+        if(errorListener.didThrow()) {
+            throw new RuntimeException(errorListener.getThrowable());
```

Comment:
Same thing, need a message here to provide context - e.g. "Unable to delete jobs, cause: "